hls4ml/model/optimizer/passes/batchnorm_opt.py

```diff
@@ -0,0 +1,269 @@
+import numpy as np
+
+from hls4ml.model.layers import BatchNormalization, BatchNormOnnx, Constant
+from hls4ml.model.optimizer import OptimizerPass
+from hls4ml.model.quantizers import QuantNodeQuantizer
+from hls4ml.model.types import FixedPrecisionType, IntegerPrecisionType, UnspecifiedPrecisionType
+
+_base_attributes = ('epsilon', 'n_in', 'n_filt')
+
+
+class BatchNormOnnxConstantParameters(OptimizerPass):
+    """Remove Constant from the BatchNormalization node parameters (but not input[0])"""
+
+    def match(self, node):
+        is_match = isinstance(node, BatchNormOnnx) and any(node.inputs[1:])
+
+        return is_match
+
+    def transform(self, model, node):
+        """
+        Remove Constant from the BatchNormalization node parameters (but not input[0])
+
+        TODO:  Currently the quantizers are not actually used by the underlying layer.
+        """
+
+        if not (len(node.inputs) == 5 and all(node.inputs)):
+            raise ValueError(f'All {len.node.inputs} BatchNormOnnnx inputs need to be defined')
```

Comment:
Not sure I understand the error message here, shouldn't it just read "All 5 BatchNormOnnnx inputs need to be defined" since it will also throw the error if all inputs are defined, but their number does not equal 5?

Reply:
I think they need to be defined and valid. The values are not optional, according to my understanding of the ONNX description. This optimizer reads those inputs, which should be Constant nodes (potentially quantized--that's why the quant-constant folding is scheduled earlier), and replaces them with '', to not shift the index of the inputs. So after this optimizer, `any(node.inputs[1:])` should return false.

Reply:
I was wondering more about the number of inputs. We check explicitly that there are 5 of them, but the error message will not tell you that the number is wrong if that check fails. So if there are 4 inputs for some reason, the check will fail, but the error message will tell you that "All 4 BatchNormOnnnx inputs need to be defined". So my comment was more about how the error message should tell you why the check actually failed.

Reply:
I see. I need to change the error message.